colcon-build/action.yaml

```diff
@@ -0,0 +1,70 @@
+name: colcon-build
+description: ""
+
+inputs:
+  rosdistro:
+    description: ""
+    required: true
+  target-packages:
+    description: ""
+    required: true
+  build-depends-repos:
+    description: ""
+    required: false
+  token:
+    description: ""
+    required: false
+    default: ${{ github.token }}
+
+runs:
+  using: composite
+  steps:
+    - name: Show target packages
+      run: |
+        echo "target packages: ${{ inputs.target-packages }}"
+      shell: bash
+
+    - name: Install pip for rosdep
+      run: |
+        sudo apt-get -yqq update
+        sudo apt-get -yqq install python3-pip
+      shell: bash
+
+    - name: Clone dependency packages
+      if: ${{ inputs.build-depends-repos != ''}}
+      run: |
+        git config --local --unset-all http.https://github.com/.extraheader || true
+        git config --global url.https://x-access-token:${{ inputs.token }}@github.com.insteadOf 'https://github.com'
```

Comment:
Could you use `set-git-config` action?
https://github.com/autowarefoundation/autoware-github-actions/tree/tier4/proposal/set-git-config

Reply:
Thank you for your suggestion! b84ad24